packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -796,6 +794,9 @@ export class Project extends ProjectBase {
     if (props.encryptionKey) {
       this.encryptionKey = props.encryptionKey;
     }
+
+    // bind
+    this.buildImage.binder?.bind(this, this, {});
```

Comment:
How about getting as close as possible to the final desired state?

If there is a bind, call it, and the bind should return all the relevant image properties. Otherwise use the properties from the current class. 

That, instead of mixing the two as is happening right now.

Reply:
I'm not sure I want to do that. That would mean a lot more complicated migration from `1.x` to `2.x` for users who have their own implementation of `IBuildImage`.